custom/champ/tests/__init__.py

```diff
@@ -29,25 +29,29 @@ def setUpModule():
 
     domain = create_domain('champ-cameroon')
     with override_settings(SERVER_ENVIRONMENT='production'):
+        try:
+            configs = StaticDataSourceConfiguration.by_domain(domain.name)
+            adapters = [get_indicator_adapter(config) for config in configs]
 
-        configs = StaticDataSourceConfiguration.by_domain(domain.name)
-        adapters = [get_indicator_adapter(config) for config in configs]
+            for adapter in adapters:
+                adapter.build_table()
 
-        for adapter in adapters:
-            adapter.build_table()
+            engine = connection_manager.get_engine(UCR_ENGINE_ID)
+            metadata = sqlalchemy.MetaData(bind=engine)
+            metadata.reflect(bind=engine, extend_existing=True)
+            path = os.path.join(os.path.dirname(__file__), 'fixtures')
+            for file_name in os.listdir(path):
+                with open(os.path.join(path, file_name), encoding='utf-8') as f:
+                    table_name = get_table_name(domain.name, file_name[:-4])
+                    table = metadata.tables[table_name]
+                    postgres_copy.copy_from(
+                        f, table, engine, format='csv' if six.PY3 else b'csv',
+                        null='' if six.PY3 else b'', header=True
+                    )
+        except Exception:
```

Comment:
what is this change addressing? i am guessing a failure in the test to still clean up?

Reply:
If there is a failure in the setup we still want to run teardown otherwise we could still end up with left over data. A failure in setup could be caused by bad data in the CSV files or some other error.